sailor/assetcentral/group.py

```diff
@@ -0,0 +1,219 @@
+
+        if not kwargs['id']:
+            warnings.warn(f'There are no "{member_name}" in this group!')
+            return set_class([])
+
+        return find_function(extended_filters=extended_filters, **kwargs)
+
+    def find_equipment(self, *, extended_filters=(), **kwargs):
+        """Retrieve all Equipment that are part of this group.
+
+        This is a wrapper for :meth:`sailor.assetcentral.equipment.find_equipment`
+        that limits the fetch query to members of this group.
+
+        This method supports the common filter language explained at :ref:`filter`.
+
+        Parameters
+        ----------
+        extended_filters
+            See :ref:`filter`.
+        **kwargs
+            See :ref:`filter`.
+        """
+        return self._generic_get_members('EQU', EquipmentSet, find_equipment, extended_filters, **kwargs)
+
+    def find_locations(self, *, extended_filters=(), **kwargs):
+        """Retrieve all Locations that are part of this group.
+
+        This is a wrapper for :meth:`sailor.assetcentral.location.find_location`
+        that limits the fetch query to members of this group.
+
+        This method supports the common filter language explained at :ref:`filter`.
+
+        Parameters
+        ----------
+        extended_filters
+            See :ref:`filter`.
+        **kwargs
+            See :ref:`filter`.
+        """
+        return self._generic_get_members('FL', LocationSet, find_locations, extended_filters, **kwargs)
+
+    def find_models(self, *, extended_filters=(), **kwargs):
+        """Retrieve all Models that are part of this group.
+
+        This is a wrapper for :meth:`sailor.assetcentral.model.find_models`
+        that limits the fetch query to members of this group.
+
+        This method supports the common filter language explained at :ref:`filter`.
+
+        Parameters
+        ----------
+        extended_filters
+            See :ref:`filter`.
+        **kwargs
+            See :ref:`filter`.
+        """
+        return self._generic_get_members('MOD', ModelSet, find_models, extended_filters, **kwargs)
+
+
+class GroupSet(ResultSet):
+    """Class representing a group of Groups."""
+
+    _element_type = Group
+    _method_defaults = {
+        'plot_distribution': {
+            'by': 'group_type',
+        },
+    }
+
+    def _generic_get_members(self, business_object_type, set_class, find_function, extended_filters, **kwargs):
+        member_name = set_class._element_type.__name__
+
+        if kwargs.get('id'):
+            raise RuntimeError(f'Can not specify `id` when retrieving "{member_name}" from a group.')
+
+        kwargs['id'] = set([item['businessObjectId'] for group in self.elements for item in group._members_raw
+                            if item['businessObjectType'] == business_object_type])
+        if not kwargs['id']:
+            warnings.warn(f'There are no "{member_name}" in any of the groups in this set!')
```

Comment:
You could use the DataNotFoundWarning we have for cases like this. Might not be completely accurate though. Give it a thought.

Reply:
I think I'd rather keep it for cases where the data should be coming from a remote system?